src/Docker/Runner/DataLoader/ExternallyManagedWorkspaceCredentials.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\DockerBundle\Docker\Runner\DataLoader;
+
+use Keboola\DockerBundle\Exception\ExternalWorkspaceException;
+use Keboola\StagingProvider\Provider\Credentials\DatabaseWorkspaceCredentials;
+
+readonly class ExternallyManagedWorkspaceCredentials
+{
+    private function __construct(
+        public string $id,
+        public string $type,
+        public string $password,
+    ) {
+    }
+
+    public static function fromArray(array $credentials): self
+    {
+        if (!isset($credentials['id']) || !isset($credentials['type']) || !isset($credentials['#password'])) {
+            throw new ExternalWorkspaceException(
+                'Missing required fields (id, type, #password) in workspace_credentials',
+            );
+        }
+        if ($credentials['type'] !== 'snowflake') {
+            throw new ExternalWorkspaceException(sprintf('Unsupported workspace type "%s"', $credentials['type']));
+        }
+        return new self(
+            (string) $credentials['id'],
+            $credentials['type'],
+            (string) $credentials['#password'],
+        );
+    }
+
+    public function getDatabaseCredentials(): DatabaseWorkspaceCredentials
```

Comment:
Nemelo by tu byt spis?
```suggestion
    public function getDatabaseCredentials(): CredentialsInterface
```
Predpokladam ze zahy nekdo prijde, ze je to potreba pro Bigquery?

Reply:
Tak to furt bude DatabaseWorkspaceCredentials a jestli tohle bude chtít někdo pro ABS, tak ho osobně zaškrtím.

Reply:
No vychazel jsem z toho, ze BQ credentials nededi z Database credentials, ale implementuji ten interface
![image](https://github.com/user-attachments/assets/9db7ad6f-6e87-428c-ae84-be5241ddae86)

Reply:
ah sakra, to mi ušlo

Reply:
No, vlastně mi to asi úplně neušlo https://github.com/keboola/docker-bundle/pull/763/files#diff-349458476ead9e21d264821d27c3af824fe286e26ea2425ba62ed7c73c06009cR26

Reply:
jasně, čili signature je CredentialsInterface, ale vrací to DatabaseWorkspaceCredentials, got it